presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -382,38 +399,52 @@ public Boolean visitWindowFrame(WindowFrame node, Void context)
         }
 
         @Override
-        protected Boolean visitIdentifier(Identifier node, Void context)
+        protected Boolean visitIdentifier(Identifier node, Boolean isAggregation)
         {
-            return isField(QualifiedName.of(node.getName()));
+            return isField(node, QualifiedName.of(node.getName()), isAggregation);
         }
 
         @Override
-        protected Boolean visitDereferenceExpression(DereferenceExpression node, Void context)
+        protected Boolean visitDereferenceExpression(DereferenceExpression node, Boolean isAggregation)
         {
             if (columnReferences.contains(node)) {
-                return isField(DereferenceExpression.getQualifiedName(node));
+                return isField(node, DereferenceExpression.getQualifiedName(node), isAggregation);
             }
 
             // Allow SELECT col1.f1 FROM table1 GROUP BY col1
-            return process(node.getBase(), context);
+            return process(node.getBase(), isAggregation);
         }
 
-        private Boolean isField(QualifiedName qualifiedName)
+        private Boolean isField(Expression node, QualifiedName qualifiedName, Boolean isAggregation)
         {
-            List<Field> fields = scope.getRelationType().resolveFields(qualifiedName);
-            checkState(!fields.isEmpty(), "No fields for name '%s'", qualifiedName);
-            checkState(fields.size() <= 1, "Found more than one field for name '%s': %s", qualifiedName, fields);
+            Scope scope = orderByScope.orElse(sourceScope);
+
+            ResolvedField resolvedField = scope.resolveField(node, qualifiedName);
+            if (orderByScope.isPresent() && resolvedField.getScope().equals(orderByScope.get())) {
+                if (isAggregation) {
+                    throw new SemanticException(REFERENCE_TO_OUTPUT_ATTRIBUTE_WITHIN_ORDER_BY_AGGREGATION, node, "Invalid reference to output projection attribute from ORDER BY aggregation");
+                }
+
+                return true;
+            }
+
+            if (resolvedField.getScope().equals(sourceScope)) {
```

Comment:
I think it can be false. For instance it can be resolved by outer scope (e.g: correlated column). I think it should pass the `AggregationAnalyzer` then since it's constant effectively. To be addressed in separate commit.